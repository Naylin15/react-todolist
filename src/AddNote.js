import React, { Component } from 'react';
import Note from './Note';
import { Form, Button } from 'semantic-ui-react';

class AddNote extends Component {
    constructor(props) {
        super(props);
        this.state = {value: ''};
    
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({value: event.target.value});
    }

    handleSubmit(event) {
        event.preventDefault();
        this.addNote(this.props.notesArray);
    }

    clearForm() {
        this.setState({value: ''});
    }    

    addNote(notes) {
        var newArray = notes;
        newArray.push({title: this.state.value});
        this.clearForm();
        return newArray;
    }    

    
    render() {
        return (
            <div>
        <div className="ui centered grid container shadow">
            <div className="ui container row">
                <h2 className="ui header">To-Do List</h2>
            </div>
            <div className="ui container row">
                <div className="ui center aligned form">
                    <Form onSubmit={this.handleSubmit}>
                        <Form.Field>
                            <input placeholder="What's next?" value={this.state.value} onChange={this.handleChange} />
                        </Form.Field>
                        <Button type='submit' className='ui primary button'>Submit</Button>
                    </Form>
                </div>
            </div>
        </div>
            <Note notesArray={this.props.notesArray}/>
        </div>

        )
    }
}

export default AddNote