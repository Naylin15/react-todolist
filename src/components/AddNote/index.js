import React, { Component } from 'react';
import Note from '../Note';
import { Form, Grid, Header } from 'semantic-ui-react';

class AddNote extends Component {
    constructor() {
        super();
        this.state = {value: '', notes: []};
    
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.updateState = this.updateState.bind(this);
    }

    handleChange(event) {
        this.setState({value: event.target.value});
    }

    handleSubmit(event) {
        event.preventDefault();
        this.addNote(this.state.notes);
    }

    clearForm() {
        this.setState({value: ''});
    }    

    addNote(notes) {
        var newArray = notes;
        newArray.push({title: this.state.value});
        // should I call updateState here? in order to not repeat the setState line
        this.setState({notes: newArray});
        this.clearForm();
    }

    updateState(newArray) {
        this.setState({notes: newArray});
    }
    
    render() {
        return (
            <div>
                <Grid container centered className='shadow'>
                    <Grid.Row>
                        <Header as='h2'>To-Do List</Header>
                    </Grid.Row>
                    <Grid.Row>    
                        <Form onSubmit={this.handleSubmit}>
                            <Form.Field>
                                <input placeholder="What's next?" value={this.state.value} onChange={this.handleChange} required/>
                            </Form.Field>
                            <button type='submit' className='ui primary button'>Add</button>
                        </Form>
                    </Grid.Row>
                </Grid>
                <Note notesArray={this.state.notes} updateState={this.updateState} />
        </div>

        )
    }
}

export default AddNote