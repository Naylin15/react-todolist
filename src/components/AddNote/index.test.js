import React from 'react';
import renderer from 'react-test-renderer'
import AddNote from './index';

it('renders correctly', () => {
    const tree = renderer
      .create(<AddNote />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });