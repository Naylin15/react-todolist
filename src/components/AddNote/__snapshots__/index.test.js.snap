// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="ui centered container grid shadow"
  >
    <div
      className="row"
    >
      <h2
        className="ui header"
      >
        To-Do List
      </h2>
    </div>
    <div
      className="row"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <input
            onChange={[Function]}
            placeholder="What's next?"
            required={true}
            value=""
          />
        </div>
        <button
          className="ui primary button"
          type="submit"
        >
          Add
        </button>
      </form>
    </div>
  </div>
  <div
    className="ui grid"
  />
</div>
`;
